resources/configure_sh.rb

```diff
@@ -0,0 +1,34 @@
+# Resource properties
+property :basic_opts, Hash
+property :additional_opts, Hash
+
+load_current_value do |new_resource|
+  basic_opts ::Mapr::ConfigureSh.load_options(new_resource.name, 'basic')
+  additional_opts ::Mapr::ConfigureSh.load_options(new_resource.name, 'add')
+end
+
+# The run action build a command line with the options (basic and additional) and execute it
+action :run do
+  converge_if_changed(:basic_opts, :additional_opts) do
+    name = new_resource.name
+    basic_opts = new_resource.basic_opts
+    additional_opts = new_resource.additional_opts
+    execute "MapR #{name}" do
+      command ::Mapr::ConfigureSh.build_command((basic_opts || {}).merge(additional_opts || {}))
+    end
+    # We store the options in a specific file to be able to compare for idempotency
+    ::Mapr::ConfigureSh.store_options(name, basic_opts, additional_opts)
```

Comment:
Why not using the file resource?

Reply:
Since I need the "load_options" function in the load_current_resource, I prefered to keep "symmetry" with a "store_options" fonction.

The call is inside the converge_if_changed, so idempotency is preserved